### ISSUE TITLE
Adding an output layer for MCTS exploration

### ISSUE THREAD

**Issue by Centaurea-Platinum** [lightvector/KataGo#1081]
It seems probable that adding a layer in the output tensor and letting the neural net tell the MCTS how to distribute computing power in the tree search could be more effective then the current noise based methodology? Perhaps the model can learn positions where there are likely blindspots and lean the tree search there even though the winrates of these positions are low initially.

**Comment by lightvector** [comment]
> I reason it could probably be trained as part of the self play?
I don't think you can just handwave at this and say that it gets "trained" when you're designing new algorithms. For the "A" output what is the loss function, or what specific number/numbers is it predicting, or what RL algorithm is being used to update it in response to what reward? If a model has a random allocation, then it will play poorly, sure, so then what causes the "A" output of the net to then change to an allocation that is better than random?

**Comment by lightvector** [comment]
Again, can you be more precise? What do you mean by "evaluation results" - do you mean utility, i.e. winrate and score? And can you be more precise about what you mean by train to match them? 

For example, if you:
* Record the winrate and score of various moves in various board positions of a large visits search into your dataset
* During training, have the model take as input those board positions and output predicted winrates and scores.
* And penalize the outputted winrates and scores based on how much they differ from the ones you recorded in your dataset
* Then the result will be the model will learn to look at a board position and output winrates and score values that are decent guesses of what the winrates and scores of a large visits search would be in that position.

But this means the model is outputting a prediction of winrates and scores, rather than the "A" output that you said you wanted. Is that what you want to do?

(If you're somewhat new to machine learning, and you're trying to do this kind of learning algorithm design, it may helpful to prohibit yourself from using the word "train" unless you also say specifically what you are recording as the data, what part of that data is the input to the model is, what the output is, and what the penalty or reward on that output is, to force yourself to be precise when thinking or communicating, and to make sure for yourself that you aren't skipping over a critical detail or just handwaving something to be possible when you haven't really figured out a way to do it).

I don't think I see yet any reason why any "teacher student" setup of that kind would be any better at fixing blind spots than the AlphaZero algorithm already is. AlphaZero is already a "teacher student" setup, and it already learns and fixes blind spots if they're ones that the teacher is readily capable of solving. But it doesn't learn and fix blind spots that the teacher is incapable of solving. Conceptually, the fundamental issue is still the same one - if the teacher persistently fails to find a move because the blind spot is severe enough, and the student is always being trained to match the teacher, there is nothing that teaches the student to find that move either.
